src/deluge/gui/views/session_view.cpp

```diff
@@ -2560,26 +2570,47 @@ void SessionView::transitionToSessionView() {
 	}
 	else {
 		int32_t transitioningToRow = getClipPlaceOnScreen(getCurrentClip());
-		InstrumentClip* instrumentClip = getCurrentInstrumentClip();
-		if (instrumentClip->onKeyboardScreen) {
-			keyboardScreen.renderMainPads(0xFFFFFFFF, &PadLEDs::imageStore[1], &PadLEDs::occupancyMaskStore[1], false);
-			keyboardScreen.renderSidebar(0xFFFFFFFF, &PadLEDs::imageStore[1], &PadLEDs::occupancyMaskStore[1]);
+		if (getCurrentUI() == &automationClipView) {
+			automationClipView.renderMainPads(0xFFFFFFFF, &PadLEDs::imageStore[1], &PadLEDs::occupancyMaskStore[1],
+			                                  false);
+			if (getCurrentClip()->type == CLIP_TYPE_AUDIO) {
+				audioClipView.renderSidebar(0xFFFFFFFF, &PadLEDs::imageStore[1], &PadLEDs::occupancyMaskStore[1]);
+			}
+			else {
+				instrumentClipView.renderSidebar(0xFFFFFFFF, &PadLEDs::imageStore[1], &PadLEDs::occupancyMaskStore[1]);
+			}
```

Comment:
This logic is repeated in enough places we should probably have a function for it (`renderSdiebarForCurrentClipView`, in `ui.h` probably?)

Reply:
> This logic is repeated in enough places we should probably have a function for it (`renderSdiebarForCurrentClipView`, in `ui.h` probably?)

Done. Added to clip class :)